k9mail/src/main/java/com/fsck/k9/mailstore/ProvidedTempFileBody.java

```diff
@@ -82,11 +88,11 @@ public long getSize() {
         throw new IllegalStateException("Data must be fully written before it can be read!");
     }
 
-    public File getFile() throws IOException {
+    public Uri getProviderUri(String mimeType) throws IOException {
```

Comment:
Halfway through implementing: The AttachmentInfoExtractor now needs to know about Context again, and makes the assumption out of nowhere that the File in the body is related to the DecryptedFileProvider (detail from an entirely different place in the codebase), then pass its file together with the body's encoding (previously implementation detail) to the provider.

Not really convinced of this; I don't see how forcefully avoiding `getUri` as an access method for body data is useful if we have to use `getFile` + previously unrelated implementation knowledge instead...